### PR TITLE
feat: implement cut operation in extrude sketch

### LINKED ISSUE #1503: Extend ``extrude-sketch`` operation to also allow for "cut" operation
Within SC and other CAD products at Ansys, one can draw a sketch and by default it will perform a extrude operation. However, you can also configure it to perform a "cut" operation (i.e. subtract operation). Let's enable this option as well - should be fairly straightforward.

### PR DESCRIPTION
**Pull request by RobPasMue**
## Description
Provide ``cut`` argument in ``extrude_sketch`` function to allow for subtract operations.

## Issue linked
Closes #1503 

## Checklist
- [X] I have tested my changes locally.
- [x] I have added necessary documentation or updated existing documentation.
- [X] I have followed the coding style guidelines of this project.
- [x] I have added appropriate unit tests.
- [X] I have reviewed my changes before submitting this pull request.
- [X] I have linked the issue or issues that are solved to the PR if any.
- [X] I have assigned this PR to myself.
- [X] I have added the minimum version decorator to any new backend method implemented.
- [X] I have made sure that the title of my PR follows [Conventional commits style](https://www.conventionalcommits.org/en/v1.0.0/#summary) (e.g. ``feat: extrude circle to cylinder``)

### CODE REVIEW

File: src/ansys/geometry/core/designer/component.py [hunk 1]
```diff
@@ -481,11 +496,16 @@ def extrude_sketch(
...
+        cut : bool, default: False
+            Whether to cut the extrusion from the existing component.
+            By default, the extrusion is added to the existing component.
 
         Returns
         -------
         Body
             Extruded body from the given sketch.
+        None
+            If the cut parameter is True, the function will return None.
```
Comment (PipKat) [inline src/ansys/geometry/core/designer/component.py:508]:
```suggestion
            If the cut parameter is ``True``, the function returns ``None``.
```

File: tests/integration/test_design.py [hunk 2]
```diff
@@ -2738,3 +2738,12 @@ def test_surface_body_creation(modeler: Modeler):
     assert len(design.bodies) == 6
     assert not body.is_surface
     assert body.faces[0].area.m == pytest.approx(39.4784176044 * 2)
+
+
+def test_extrude_sketch_with_cut_request(modeler: Modeler):
+    """Test the cut argument when performing a sketch extrusion (this will mimic a cut operation).
```
Comment (PipKat) [inline tests/integration/test_design.py:2744]:
```suggestion
    """Test the cut argument when performing a sketch extrusion.
    
     This method mimics a cut operation.
```

**Review by PipKat — DISMISSED** [review]
Minor edits suggested--although I see this is a draft. I only looked at it because I was copied on the PR. Sorry!

**Comment by RobPasMue** [comment]
> Minor edits suggested--although I see this is a draft. I only looked at it because I was copied on the PR. Sorry!

No problem at all - thanks @PipKat!

**Pull request merged**